src/Core/Workspace/Workspace.cs

```diff
@@ -77,10 +88,74 @@ public string CacheFolder
         /// </summary>
         public string Root { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether to skip automatically loading the .csproj from the workspace's root folder.
+        /// </summary>
+        private bool SkipAutoLoadProject { get; set; }
+
+        /// <summary>
+        /// Gets the project files to be built for this Workspace.
+        /// </summary>
+        public IEnumerable<string> ProjectFiles
+        {
+            get
+            {
+                var projectFiles = Directory.EnumerateFiles(Root, "*.csproj", SearchOption.TopDirectoryOnly);
+                if (SkipAutoLoadProject || projectFiles.Count() != 1) return Enumerable.Empty<string>();
+
+                var projectFilesToResolve = projectFiles.ToArray();
+                while (projectFilesToResolve.Any())
+                {
+                    try
+                    {
+                        Logger?.LogInformation($"Looking for project references in .csproj files: {string.Join(";", projectFilesToResolve)}");
+
+                        projectFilesToResolve = projectFilesToResolve
+                            .SelectMany(projectFile =>
+                                XDocument.Load(projectFile)
+                                    .XPathSelectElements("//ProjectReference")
+                                    .Select(element => Path.Combine(
+                                        Path.GetDirectoryName(projectFile),
+                                        element.Attribute("Include").Value.Replace('\\', Path.DirectorySeparatorChar))))
+                            .Where(projectFile => !projectFiles.Contains(projectFile))
+                            .ToArray();
+
+                        var missingFiles = projectFilesToResolve.Where(projectFile => !File.Exists(projectFile));
+                        if (missingFiles.Any())
+                        {
+                            Logger?.LogError($"Skipping invalid project references: {string.Join(";", missingFiles)}");
+                            projectFilesToResolve = projectFilesToResolve.Except(missingFiles).ToArray();
+                        }
+
+                        Logger?.LogInformation($"Adding project references: {string.Join(";", projectFilesToResolve)}");
+
+                        projectFiles = projectFiles.Concat(projectFilesToResolve);
```

Comment:
Very minor suggestion, as this is unlikely to be a performance bottleneck, but given that you're growing `projectFiles` in a loop it may be good to use a `List` instead of an array, and then replace this by `AddRange`.